### PR TITLE
client: Add missing property/method to Vehicle

### PR DESCRIPTION
**Pull request by xDavidos**
There were some shared properties in the old js module, I didn't add them here, they should be extended from `sharedVehicleClass`, right?

I try write in HandlingGetter
`ctx.Return(Handling.Create(ctx.GetContext(), ctx.GetExtraInternalFieldValue()));`
 but i expect. Should I leave it like this, or is there any way to fix it?
`no instance of function template "js::PropertyContextBase<Info>::GetExtraInternalFieldValue [with Info=v8::FunctionCallbackInfo<v8::Value>]" matches the argument list`

### CODE REVIEW

**Comment by LeonMrBonnie** [comment]
> There were some shared properties in the old js module, I didn't add them here, they should be extended from sharedVehicleClass, right?

I don't understand what you mean by that. If something is shared it should go into the shared class, yes. In old JS module a lot of stuff is duplicated on client and server when it's shared, because old JS module had no good system for shared code.

> try write in HandlingGetter ctx.Return(Handling.Create(ctx.GetContext(), ctx.GetExtraInternalFieldValue())); but i expect. Should I leave it like this, or is there any way to fix it?

Just remove that getter for now until we actually added the Handling class.
Also, for alt:V entities `GetExtraInternalFieldValue()` isn't used, that's only used for classes that don't use the alt:V entity system. For alt:V entities use `ctx.GetThisObject<Type>();`


Also I don't see a reason why you added that generic getter/setter for the wheel methods? You can just use the normal method wrapper we use everywhere.

File: client/src/classes/Vehicle.cpp [hunk 1]
```diff
@@ -1,10 +1,222 @@
...
+    tpl.Property<&alt::IVehicle::GetSpeedVector>("speedVector");
+    tpl.Property<&alt::IVehicle::IsHandlingModified>("isHandlingModified");
+    tpl.Property<&alt::IVehicle::GetLightsIndicator, &alt::IVehicle::SetLightsIndicator>("indicatorLights");
+    tpl.Property<&alt::IVehicle::GetSeatCount>("seatCount");
+    tpl.Property<&alt::IVehicle::GetOccupiedSeatsCount>("occupiedSeatsCount");
+    tpl.Property<&alt::IVehicle::IsTaxiLightOn>("TaxiLightOn");
+    tpl.Property<&alt::IVehicle::GetEngineTemperature, &alt::IVehicle::SetEngineTemperature>("engineTemperature");
+    tpl.Property<&alt::IVehicle::GetFuelLevel, &alt::IVehicle::SetFuelLevel>("fuelLevel");
+    tpl.Property<&alt::IVehicle::GetOilLevel, &alt::IVehicle::SetOilLevel>("oilLevel");
+    tpl.Property<&alt::IVehicle::GetEngineLightState, &alt::IVehicle::SetEngineLightState>("engineLight");
```
Comment (LeonMrBonnie) [inline client/src/classes/Vehicle.cpp:197]: It'd be better to add `state` to all of those properties to match their SDK name.

File: client/src/classes/Vehicle.cpp [hunk 2]
```diff
@@ -1,10 +1,222 @@
...
+    tpl.Property<&alt::IVehicle::GetCurrentGear, &alt::IVehicle::SetCurrentGear>("gear");
+    tpl.Property<&alt::IVehicle::GetMaxGear>("maxGear");
+    tpl.Property<&alt::IVehicle::GetCurrentRPM, &alt::IVehicle::SetCurrentRPM>("rpm");
+    tpl.Property<&alt::IVehicle::GetWheelsCount>("wheelsCount");
+    tpl.Property<&alt::IVehicle::GetSpeedVector>("speedVector");
+    tpl.Property<&alt::IVehicle::IsHandlingModified>("isHandlingModified");
+    tpl.Property<&alt::IVehicle::GetLightsIndicator, &alt::IVehicle::SetLightsIndicator>("indicatorLights");
+    tpl.Property<&alt::IVehicle::GetSeatCount>("seatCount");
+    tpl.Property<&alt::IVehicle::GetOccupiedSeatsCount>("occupiedSeatsCount");
+    tpl.Property<&alt::IVehicle::IsTaxiLightOn>("taxiLightOn");
```
Comment (LeonMrBonnie) [inline client/src/classes/Vehicle.cpp:193]: For properties which only have a getter and no setter you should use the `is` prefix like `isTaxiLightOn`

**Comment by LeonMrBonnie** [comment]
I still don't see a reason why you manually added bindings for the wheel methods, you could just use `tpl.Method` with the template arg

**Comment by xDavidos** [comment]
> I still don't see a reason why you manually added bindings for the wheel methods, you could just use `tpl.Method` with the template arg

Then I don't understand at all when to use template arg and when to use manually added bindings. Here we use manually added bindings because GetValueAsHash?
https://github.com/altmp/altv-js-module-v2/blob/86857c56247a58ede63313753100e000dd01461c/shared/src/classes/SharedPlayer.cpp#L4

**Comment by xDavidos** [comment]
I hope I understood everything correctly and used manually bindings where it is really needed😉

**Comment by LeonMrBonnie** [comment]
Yes, correct. Because we use GetArgAsHash there we need manual bindings, because the automatic bindings don't know that it is a hash and then would only accept number.
For most functions that just accept simple types that are convertible by the module, no manual bindings are needed at all.

**Pull request merged**